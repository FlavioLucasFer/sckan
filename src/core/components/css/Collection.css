.collection-item {
	font-size: 20px;
	cursor: pointer;
}

.collection-item.active {
	cursor: default;
}