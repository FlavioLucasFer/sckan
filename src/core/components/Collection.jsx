import './css/Collection.css';
import React, { useState } from 'react';
import PropTypes from 'prop-types';

Collection.propTypes = {
	items: PropTypes.arrayOf(Object).isRequired,
	identifier: PropTypes.string.isRequired,
	onSelect: PropTypes.func,
};

function Collection(props) {
	const {
		items,
		identifier,
		onSelect,
	} = props;

	const [activeItem, setActiveItem] = useState(-1);

	function isActive(index) {
		return index === activeItem;
	}

	function handleListItemClick(index) {
		setActiveItem(index);

		if (onSelect)
			onSelect(items[index]);
	}

	return (
		<ul className="collection">
			{items.map((e, i) => (
				<li key={i} 
					id={i}
					className={`collection-item ${isActive(i) && 'active'}`}
					onClick={() => handleListItemClick(i)}>
					{e[identifier]}
				</li>
			))}
		</ul>
	);
}

export default Collection;