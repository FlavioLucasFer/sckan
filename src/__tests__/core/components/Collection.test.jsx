import React from 'react';
import { shallow } from 'enzyme';

import Collection from 'core/components/Collection';

describe('Testing Collection core component', () => {
	const items = [
		{
			id: 1213,
			companyName: 'Testing',
			tradingName: 'Hard code test',
			federalDocument: '111.111.111-11',
			email: 'mail@email.com',
		}, {
			companyName: 'bbb',
			id: 1345,
			federalDocument: '131.131.131-13',
			email: 'amail123@email.commmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm',
			tradingName: 'absdbstsh',
		}, {
			companyName: 'ssahsas',
			id: 1818,
			federalDocument: '121.121.121-12',
			email: 'mail123@email.com',
			tradingName: 'Hash',
		},
	];

	it('should render correctly', () => {
		const wrapper = shallow(
			<Collection items={items}
				identifier="companyName" />
		);
		
		expect(wrapper.debug()).toMatchSnapshot();
	});
	
	it('should select list item on click', () => {
		const wrapper = shallow(
			<Collection items={items}
				identifier="companyName"
				onSelect={onSelect} />
		);

		let selectedItem;

		function onSelect(item) {
			selectedItem = item;
		}

		wrapper.find({ id: 1 }).simulate('click');

		expect(selectedItem).toBe(items[1]);
		expect(wrapper.find({ id: 1 }).hasClass('active')).toBeTruthy();
		expect(wrapper.find({ id: 0 }).hasClass('active')).toBeFalsy();
	});
});