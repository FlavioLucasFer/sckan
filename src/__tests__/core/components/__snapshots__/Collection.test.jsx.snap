// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Collection core component should render correctly 1`] = `
"<ul className=\\"collection\\">
  <li id={0} className=\\"collection-item false\\" onClick={[Function: onClick]}>
    <span className=\\"item-id grey-text text-darken-3\\">
      #
      1213
    </span>
    Testing
  </li>
  <li id={1} className=\\"collection-item false\\" onClick={[Function: onClick]}>
    <span className=\\"item-id grey-text text-darken-3\\">
      #
      1345
    </span>
    bbb
  </li>
  <li id={2} className=\\"collection-item false\\" onClick={[Function: onClick]}>
    <span className=\\"item-id grey-text text-darken-3\\">
      #
      1818
    </span>
    ssahsas
  </li>
</ul>"
`;
